tcms/bugs/tests/test_views.py

```diff
@@ -0,0 +1,89 @@
+# pylint: disable=too-many-ancestors
+from http import HTTPStatus
+from django.urls import reverse
+
+from tcms.tests.factories import BugFactory
+from tcms.tests import BaseCaseRun, user_should_have_perm
+from tcms.utils.permissions import initiate_user_with_default_setups
+
+
+class TestBugStatusChange(BaseCaseRun):
+    """Test the possible bug status changes.
+
+    Cases:
+    * Closing an open bug.
+    * Reopenning a closed bug.
+    * Try to reopen an already open bug.
+    * try to close an already closed bug.
+    """
+
+    @classmethod
+    def setUpTestData(cls):
+        super().setUpTestData()
+        initiate_user_with_default_setups(cls.tester)
+
+        cls.bug = '1'
+        cls.text = 'Test Comment'
+        cls.action = 'close'
+        cls.data = {
+            'bug': cls.bug,
+            'text': cls.text,
+            'action': cls.action
+        }
+        cls.comment_bug_url = reverse('bugs-comment')
+
+        user_should_have_perm(cls.tester, 'bugs.change_bug')
+
+    def test_close_an_openned_bug(self):
+        bug = BugFactory(status=True)
+        edit_bug_data = self.data.copy()
+        edit_bug_data['bug'] = bug.pk
+        edit_bug_data['action'] = 'close'
+        redirect_url = reverse('bugs-get', args=[bug.pk])
+        response = self.client.post(self.comment_bug_url, edit_bug_data)
+        bug.refresh_from_db()
+
+        # Dont know why server returns 302.
```

Comment:
because it posts a comment and then redirects you to `bugs-get`

Reply:
So its ok to expect 302. I have removed the comment.